src/components/CellsContainer/CellsContainer.svelte

```diff
@@ -0,0 +1,201 @@
+<script>
+  import { onMount, onDestroy } from 'svelte';
+  import { select } from 'd3';
+  import { createSvg, getId, createRectsData, setAttributes } from './helpers';
+
+  export let count = 2;
+  export let size = 100;
+  export let gap = 1;
+  export let getAdditionalRectProps;
+  export let tooltipBackground = '#4a5568';
+  export let tooltipColor = 'MintCream';
+  export let color = 'SteelBlue';
+  export let hoverColor = 'IndianRed';
+  export let margin = { top: 20, left: 20, right: 20, bottom: 20 };
+
+  let containerRef = null;
+  let svg = null;
+  let g = null;
+  let data = [...Array(count)];
+  let containerWidth = 0;
+  let height = 0;
+  let innerWidth = 0;
+
+  $: {
+    if (containerWidth) {
+      innerWidth = containerWidth - margin.left - margin.right;
+
+      const rectSize = size + gap * 2;
+      const colsAmount = Math.floor(innerWidth / rectSize);
+      const rowsAmount = Math.ceil(count / colsAmount);
+
+      height = rectSize * rowsAmount + margin.top + margin.bottom;
+
+      data = createRectsData({
+        containerWidth: innerWidth,
+        count,
+        size,
+        gap,
+        getAdditionalRectProps,
+        color,
+        hoverColor
+      });
+
+      updateSvgElementsSizes();
+      updateRects();
+    }
+  }
+
+  const keyGetter = (key) => (r) => r[key];
```

Comment:
is it possible to use lodash `get` here somehow?

Reply:
I don't think that we need it.
It will be `const keyGetter = (key) => (r) => get(r, key);` and I don't think it worth.

Reply:
it is worth because it's undefined safe, what if r will be empty? then you will get an error

Reply:
What if I do `const keyGetter = (key) => (r) => r && r[key];`

Reply:
But I think it's ok to use lodash get. If we need to change something in the future (chaining, for example), it will be easier with it.

Reply:
Everything becomes better with TS! )
Let's take the last segment
`(r) => r && r[key];`
we assume that `r` will be an object, where the key will be string and value will be string, and there will be no other variants like a nested object, so in TS we can express it like `{[key: string]: string}`
Next, by the requirements we want (but we can't know for sure) that the result will be a string, right?
so we can express it like this
`(r: {[key: string]: string}): string => r && r[key];`
So here is my question, what possible types can such function return in real?

Reply:
It can return any type now. We don't know exactly what is inside.

Reply:
yep, and with `get` at least it will be null or boolean safe, it could be only value from object or undefined